test/metabase/driver/postgres/test_data.clj

```diff
@@ -0,0 +1,257 @@
+(ns metabase.driver.postgres.test-data
+  (:require
+   [clojure.java.io :as io]
+   [clojure.java.jdbc :as jdbc]
+   [clojure.string :as str]
+   [java-time.api :as t]
+   [metabase.test.data.sql :as sql.tx]
+   [metabase.test.data.sql-jdbc.spec :as spec]
+   [metabase.test.test-data :as test.test-data]
+   [metabase.util :as u]
+   [metabase.util.date-2 :as u.date]
+   [metabase.util.log :as log]
+   [metabase.util.malli :as mu]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private database-sql
+  (str/join
+   \newline
+   ["DO $$ BEGIN"
+    "  PERFORM pg_terminate_backend(pg_stat_activity.pid)"
+    "  FROM pg_stat_activity"
+    "  WHERE pid <> pg_backend_pid()"
+    "    AND pg_stat_activity.datname = 'test-data';"
+    "END $$;"
+    ""
+    "-----"
+    ""
+    "DROP DATABASE IF EXISTS \"test-data\";"
+    ""
+    "-----"
+    ""
+    "CREATE DATABASE \"test-data\";"]))
+
+#_{:clj-kondo/ignore [:discouraged-var]}
+(mu/defn ^:private print-field-sql
```

Comment:
As another way to make this easier to debug, the underlying methods just print stuff to `*out*` using `print` and friends. This makes it super easy to debug them in the REPL:

```clj
(print-field-sql :postgres (test.test-data/schema "venues" "id"))
;; => "id" integer PRIMARY KEY NOT NULL
;; => nil

(print-table-sql :postgres (test.test-data/schema "venues"))
;; => CREATE TABLE "venues" (
;; =>   "id" integer PRIMARY KEY NOT NULL,
;; =>   "name" text,
;; =>   "category_id" integer REFERENCES "categories" ("id"),
;; =>   "latitude" float,
;; =>   "longitude" float,
;; =>   "price" integer
;; => );
;; => nil
```

When we generate the actual artifacts we just bind `*out*` to a `java.io.FileWriter` for the file we want to write things to.